### ISSUE TITLE
Dota 2 Hangs on Vulkan on Linux (Debian 12 on Wayland) after 10th Anniversary Update

### ISSUE THREAD

**Issue by helix-d2** [ValveSoftware/Dota2-Gameplay#10781]
### Description

Since the 10th anniversary update, Dota 2 just hangs either on start screen (after splash screen), or randomly after 10-15 seconds. I am on Debian 12 running KDE 5.27.5 on Wayland. Please let me know if there is anything needed from me to diagnose the problem.

**Comment by JonesN3** [comment]
Same for me. Fedora 37 running KDE 5.27.4 on Wayland.

**Comment by pyaesone-gipsy** [comment]
Same for me, ubuntu 23.04 with nvdia graphic

**Comment by iluha168** [comment]
Same for me, Debian 12, KDE 5.27.7 on X11 with NVIDIA GeForce GTX 1050 Ti (Driver version 525.125.06)

**Comment by JonesN3** [comment]
I updated my drivers and the issue has been resolved!

**Comment by helix-d2** [comment]
@JonesN3 which version did you update your drivers to and what's your GPU? My NVIDIA Geforce GTX 1050 Ti (Notebook) is currently using driver version 525.125.06. So far, no driver updates available yet for Debian 12.

**Comment by pyaesone-gipsy** [comment]
> @JonesN3 which version did you update your drivers to and what's your GPU? My NVIDIA Geforce GTX 1050 Ti (Notebook) is currently using driver version 525.125.06. So far, no driver updates available yet for Debian 12.

Agree, no way to update the driver. only valve need to check and release the patch.

**Comment by LokeLok** [comment]
Open a new issue over at: https://github.com/ValveSoftware/Dota-2 to get help with Linux issues.

**Comment by neimovirne** [comment]
same for me on ubuntu 23.04 X/Waayland

**Comment by Sectoid** [comment]
Experiencing the same issue. Debian 12, nVidia GTX1070, driver 525.125, Vulkan API.

Dota hangs completely on start, seems like on attempt to show new cosmetics. dmesg log:

[ 124.120083] NVRM: GPU at PCI:0000:08:00: GPU-ec27b75d-cf7d-4f18-dc30-25947e03b260
[ 124.120088] NVRM: Xid (PCI:0000:08:00): 32, pid=9769, name=dota2, Channel ID 00000066 intr 00040000
[ 124.120413] NVRM: Xid (PCI:0000:08:00): 32, pid=9769, name=dota2, Channel ID 00000066 intr 00800000

**Comment by punished-scrappy** [comment]
Same here, Ubuntu 22.04, Nvidia 1050m on 535.86, X11.

It seems to be tied to fullscreen for me -- the game hangs if and only if it's launched in fullscreen. Worked around it by adding -sw to launch options and then switching to fullscreen after hitting the main menu. Haven't tested much but I can play a bot match just fine like this

**Comment by JonesN3** [comment]
@helix-d2 @pyaesone-gipsy I upgraded to 535.98. I'm on fedora it's in the default repositories. Cannot guarantee that is what solved it since i did a full system update, but I can't imagine anything else.

**Comment by Ank1x** [comment]
> Same here, Ubuntu 22.04, Nvidia 1050m on 535.86, X11.
> 
> It seems to be tied to fullscreen for me -- the game hangs if and only if it's launched in fullscreen. Worked around it by adding -sw to launch options and then switching to fullscreen after hitting the main menu. Haven't tested much but I can play a bot match just fine like this

I tested it and you're right, launching with -sw did work for me, but after switching to desktop-friendly fullscreen, it hanged again. I am still on 525.125.06 though, still hoping for a graphics driver update but they're slower on Debian so it might take some time. In the meantime, let's make this issue visible to Dota 2 devs so they can try to fix it from their end.

**Comment by helix-d2** [comment]
> @helix-d2 @pyaesone-gipsy I upgraded to 535.98. I'm on fedora it's in the default repositories. Cannot guarantee that is what solved it since i did a full system update, but I can't imagine anything else.

Nice, No updates from Debian repos yet. Fedora seems faster with the updates.

**Comment by Sectoid** [comment]
Upgrading nvidia-driver to 535.98 solved that for me on Debian 12.

**Comment by klavitopalm** [comment]
I had the same issue on Ubuntu 22.04.3 LTS. I was running nvidia driver 525.xx. I updated to nvidia drivers 535.85.05 and now it's fine again. I used the official repos and did not package it myself.

**Comment by BrianIsKool** [comment]
Valve, as always, don't give a shit about Linux users...

**Comment by hanhbd1** [comment]
same with me. auto crash while play, and got an abadon because of that

**Comment by constlo** [comment]
same here. Crashing on start with Pop_os 22.04

**Comment by vuongtuha** [comment]
I came late, but hanging on Debian Sid (trixie) with Wayland session, and ofc Nvidia 545.23
I tried not only Gnome, KDE, but also Hyprland, Sway. There's only one way for me now is opening another x11 session to play game only 
Not forget that i put SDL_VIDEO_DRIVER=x11 already
